packages/release/preconfigured.target

```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Bottlerocket dynamic pre-configuration
+AllowIsolate=yes
+After=basic.target
+Requires=basic.target settings-applier.service
```

Comment:
I used `settings-applier` since it is the last "leaf" in the dependency tree, all the services required by `preconfigured.target` are effectively "successful" if `settings-applier` exits successfully

Reply:
This goes along with my comment about disliking transitive dependencies: it's hard to trust and reason about `settings-applier` being correct here. What if we add some other service that starts after settings-applier?

I'd prefer that targets mostly just express dependencies on other targets in their own units, and that we fill in the dependency graph in service units instead.